src/consultations/consultations.controller.ts

```diff
@@ -106,6 +106,59 @@ export class ConsultationsController {
     })
   }
 
+  @JwtAuth()
+  @Post(':id/leave')
+  async leave(
+    @Param('id', ParseIntPipe) id: number,
+    @CurrentUser() user: UserEntity,
+  ): Promise<ParticipationEntity> {
+    const participation = await this.participationService.findOne(id, user.id)
+    if (participation === null) {
+      throw new HttpException(
+        'Nem vagy résztvevője ennek a konzultációnak!',
+        HttpStatus.BAD_REQUEST,
+      )
+    }
+
+    return this.participationService.remove(participation.id)
+  }
+
+  @JwtAuth()
+  @Patch(':id/rate')
+  async editRating(
+    @Param('id', ParseIntPipe) id: number,
+    @CurrentUser() user: UserEntity,
+    @Body() { ratedUserId, ...ratingDto }: CreateRatingDto,
+  ): Promise<RatingEntity> {
+    const participation = await this.participationService.findOne(id, user.id)
+    if (participation === null) {
+      throw new HttpException(
+        'Nem vagy résztvevője ennek a konzultációnak!',
+        HttpStatus.BAD_REQUEST,
+      )
+    }
+
+    const presentation = await this.presentationService.findOne(id, ratedUserId)
+    if (presentation === null) {
+      throw new HttpException(
+        'Ez a felhasználó nem előadója ennek a konzultációnak!',
+        HttpStatus.BAD_REQUEST,
+      )
+    }
+
+    const rating = await this.ratingService.findByIds(presentation.id, user.id)
```

Comment:
`ratingsService`'s `findByIds` method finds a rating by presentation and participation id, not by user id. I think you didn't notice this because `user.id` was, by chance equal to `participation.id`, but that obviously won't always be like that.